ptkit/build.gradle

```diff
@@ -1,50 +1,57 @@
-apply plugin: 'com.android.library'
-
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
-
-group = 'com.github.pintraveler'
-
-repositories {
-    mavenCentral()
+plugins {
+    id 'com.android.library'
+    id 'org.jetbrains.kotlin.android'
+    id 'kotlin-android-extensions'
+    id 'maven-publish'
 }
 
 android {
-    compileSdkVersion 30
-    buildToolsVersion "29.0.3"
+    compileSdk 32
+
     defaultConfig {
-        minSdkVersion 26
-        targetSdkVersion 30
+        minSdk 21
+        targetSdk 32
+
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles "consumer-rules.pro"
     }
+
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
 }
 
 dependencies {
-    // So Firebase dependencies move together
-    implementation platform('com.google.firebase:firebase-bom:25.12.0')
+    implementation 'androidx.recyclerview:recyclerview:1.2.1'
 
-    // Firebase dependencies
-    implementation 'com.google.firebase:firebase-core:18.0.2'
+    implementation platform('com.google.firebase:firebase-bom:25.12.0')
     implementation 'com.google.firebase:firebase-auth-ktx'
     implementation 'com.google.firebase:firebase-firestore-ktx'
 
-    //
-    implementation 'androidx.recyclerview:recyclerview:1.1.0'
-
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'androidx.appcompat:appcompat:1.2.0'
-    implementation 'androidx.core:core-ktx:1.3.2'
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'androidx.test:runner:1.3.0'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
-
     implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.4.1'
     implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-play-services:1.3.7'
+
 }
+
+afterEvaluate {
+    publishing {
+        publications {
+            release(MavenPublication) {
+                from components.release
+                groupId = 'com.github.pintraveler'
+                artifactId = 'ptkit'
+                version = '0.0.4'
```

Comment:
bunu manuel olarak degistirmemiz gerekiyor diye anliyorum? Githubda bir tag'e vs. baglamak kolay mi?

Reply:
sanırım değiştirmeden de tag'de "0.0.5" olarak belirleyebiliyoruz bir sorun olmuyor. ancak code clean olması için değiştirmek daha iyi olur diye düşünüyorum. bildiğim kadarıyla otomatik olmuyor manuel ayarlamak gerekiyor. otomatiğini bulursak süper olur